only use major version to specify python version

oToToT opened the pull request:
I think it is more flexible to use only major version of python. Also, miku seems only check the major version of python, so it might not be necessary to specify minor version in the code.

oToToT committed:
only use major version to specify python version

TaWeiTu commented on src/batchjudge.cpp:
```diff
@@ -52,9 +52,9 @@ int BatchJudge(int pid, int td, int boxid, int tl, int ml, int ol, int testee,
+    sandboxExec(boxid, opt, {"/usr/bin/env", "python2", "main.pyc"});
   } else if (lang == "python3") {
-    sandboxExec(boxid, opt, {"/usr/bin/env", "python3.7", "main.pyc"});
+    sandboxExec(boxid, opt, {"/usr/bin/env", "python3", "main.pyc"});
```
I think something like
```cpp
if (lang.startswith("python")) {
  sandboxExec(..., {"/usr/env/bin", lang, ...});
} else {
  // C++
}
```
is more elegant.

oToToT replied:
How about something like this?  
use `lang.find_first_of("python") == 0` as `lang.starts_with`.

Maybe we could change this when C++20 being released officially.

TaWeiTu replied:
LGTM

oToToT committed:
simplify python2 and python3 stuff

use lang.find_first_of to check whether the language is python,
and do python things

oToToT committed:
Merge branch 'main' into py-major-ver

TaWeiTu reviewed (approved):
LGTM
